File: Frontend/SPFrontend/src/app/components/models/ExamDTO.ts
import {User} from "./User";
import {ModuleDTO} from "./ModuleDTO";

export class ExamDTO {
  exam_id: number;
  title: string;
  creation_date: string ;
  exam_date: number;
  status: any;
  total_points: number;
  module: ModuleDTO;

  constructor(exam_id,title,creation_date,exam_date,status,total_points,module) {
    this.exam_id = exam_id;
    this.title = title;
    this.creation_date = creation_date;
    this.exam_date = exam_date;
    this.status = status;
    this.total_points= total_points;
    this.module = module;
  }

}

export class CreateExamDTO {
  title: string;
  creation_date: number ;
  exam_date: number;
  status: string;
  module: ModuleDTO;

  constructor(title:string,creation_date:number,exam_date:number,user:User,module:ModuleDTO) {
    this.title = title;
    this.creation_date = creation_date;
    this.exam_date = exam_date;
    this.module = module;
    this.status = "in_creation";
  }

}

File: Backend/src/main/java/de/hft_stuttgart/winf/proj2/sp/backend/dto/ExamDto.java
package de.hft_stuttgart.winf.proj2.sp.backend.dto;

import de.hft_stuttgart.winf.proj2.sp.backend.util.Column;
import lombok.*;

import java.util.Date;


/**
 * This Class is for defining a Exam and correlating the correct DB-field to it
 */
@AllArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class ExamDto {

    @Column("exam_id")
    private Integer exam_id;
    @Column("title")
    private String title;
    @Column("creation_date")
    private Date creation_date;
    @Column("exam_date")
    private Date exam_date;
    @Column("status")
    private String status;
    @Column("module_id")
    private Integer moduleId;
    @Column("total_points")
    private Integer totalPoints;

    @Setter
    private ModuleDto module;


    public ExamDto(){


    }
}


